src/heaps/mutable_binary_heap.jl

```diff
@@ -251,6 +251,36 @@ function update!(h::MutableBinaryHeap{T}, i::Int, v) where T
         _heap_bubble_down!(comp, nodes, nodemap, nd_id)
     end
 end
+"""
+    delete!{T}(h::MutableBinaryHeap{T}, i::Int)
+
+Delete the element present at index `i` from heap `h` .    
+"""
+function delete!(h::MutableBinaryHeap{T}, i::Int) where T
+    nodes = h.nodes
+    nodemap = h.node_map
+    comp = h.comparer
+
+    nd_id = nodemap[i]
+    rt = nodes[nd_id]
+    v0 = rt.value 
+    @inbounds nodemap[rt.handle] = 0
+
+    if length(nodes) == 1
+        #clear
+        empty!(nodes)
+    else
+        #place last node to removed one
+        @inbounds nodes[nd_id] = new_rt = pop!(nodes)
+        @inbounds nodemap[new_rt.handle] = 1
+
+        if length(nodes) > 1
+            _heap_bubble_down!(comp, nodes, nodemap, nd_id)
+        end
+    end
+    v0
+end
```

Comment:
I believe, and you should confirm this,
The function that deleted an element at index `i`, and returns that element,
is called 
`pop!(collection, i)`.
for arrays and dicts.

`delete!(collection, i)` is the same but it always returns `nothing`.

Do check this though

Reply:
The `delete!(Dict, key)` and `deleteat!(Vector, i)` returns the remaining collection after deleting the element specified - https://github.com/JuliaLang/julia/blob/903644385b91ed8d95e5e3a5716c089dd1f1b08a/base/array.jl#L819-L834

Reply:
So finally what should it be returning? The final container? or `nothing`? Any comments?

Reply:
You should return the final container.
Sorry, i misremembered